src/main/webapp/mapScript.js

```diff
@@ -25,21 +32,83 @@ function initMap() {
 function geocodeAddress(geocoder, resultsMap) {
     const address = document.getElementById('address').value;
     geocoder.geocode({'address': address}, (results, status) => {
-    if (status === 'OK') {
-        resultsMap.setCenter(results[0].geometry.location);
+        if (status === 'OK') {
+            resultsMap.setCenter(results[0].geometry.location);
```

Comment:
If this is doing what I think it's doing (and I might be misunderstanding), it's used to relocate the map to a certain city/location near the user. would it be possible to use the user's location automatically? if not, I'm wondering why you have to indicate an index for results. are you expecting more than one location to be returned?

Reply:
For this section it is actually taking the address that the user inputs in the search bar and centering the map at that location. The status is to verify that the address/location is valid. The results[0] is the location that the map can use (long/lat) to shift the map to that place.